channels/channel.be/vier/chn_vier.py

```diff
@@ -596,23 +596,27 @@ def __extract_stream_collection(self, item, video_info):
                 item.isDrmProtected = True
                 item.metaData["drm"] = drm_key
                 return
-
-            streams = stream_collection["streams"]
-            duration = stream_collection["duration"]
-            if duration:
-                item.set_info_label(MediaItem.LabelDuration, duration)
-            for stream in streams:
-                proto = stream["protocol"]
-                stream_url = stream["url"]
-                if proto == "dash":
-                    stream = item.add_stream(stream_url, 1550 if prefer_widevine else 1450)
-                    Mpd.set_input_stream_addon_input(stream)
-                elif proto == "hls":
-                    stream = item.add_stream(stream_url, 1500)
-                    M3u8.set_input_stream_addon_input(stream)
-
-                if "/geo" in stream_url:
-                    item.isGeoLocked = True
+            
+            if "duration" in stream_collection:
+                item.set_info_label(MediaItem.LabelDuration, stream_collection["duration"])
+
+            # No longer seems to be used, fallback to long-form.
+            if "streams" in stream_collection:
+                streams = stream_collection["streams"]
+                for stream in streams:
+                    proto = stream["protocol"]
+                    stream_url = stream["url"]
+                    if proto == "dash":
+                        stream = item.add_stream(stream_url, 1550 if prefer_widevine else 1450)
+                        Mpd.set_input_stream_addon_input(stream)
+                    elif proto == "hls":
+                        stream = item.add_stream(stream_url, 1500)
+                        M3u8.set_input_stream_addon_input(stream)
+
+                    if "/geo" in stream_url:
+                        item.isGeoLocked = True
```

Comment:
Ik heb geen serie meer kunnen vinden waar dit nog in zit, dus het kan eventueel ook weg.
Dit zit er nog in:
```json
'streamCollection': {
   'adInsertion': 'ssai',
   'drmProtected': False,
   'duration': 2830,
   'status': 'available'
},
```

Reply:
I always prefer to not break any backward compatibility. Some broadcasters keep old streams somewhere with older content.